datafusion/spark/src/function/bitwise/bitwise_not.rs

```diff
@@ -64,8 +66,21 @@ impl ScalarUDFImpl for SparkBitwiseNot {
         &self.signature
     }
 
-    fn return_type(&self, arg_types: &[DataType]) -> Result<DataType> {
-        Ok(arg_types[0].clone())
+    fn return_type(&self, _arg_types: &[DataType]) -> Result<DataType> {
+        internal_err!("SparkBitwiseNot: return_type() is not used; return_field_from_args() is implemented")
+    }
+
+    fn return_field_from_args(&self, args: ReturnFieldArgs) -> Result<FieldRef> {
+        if args.arg_fields.len() != 1 {
+            return plan_err!("bitwise_not expects exactly 1 argument");
+        }
+
+        let input_field = &args.arg_fields[0];
+
+        let out_dt = input_field.data_type().clone();
+        let out_nullable = input_field.is_nullable();
```

Comment:
You should also check for scalar None.
See https://github.com/apache/datafusion/pull/19242/changes#diff-cd6bd8d074f37d2c7815016e857791d6f9cb81f981110a218e59d9bd3b27f6d4R89-R94

And add a test case similar to https://github.com/apache/datafusion/pull/19242/changes#diff-cd6bd8d074f37d2c7815016e857791d6f9cb81f981110a218e59d9bd3b27f6d4R174-R181